pkg/relayer/relayminer.go

```diff
@@ -134,3 +136,48 @@ func (rel *relayMiner) ServePprof(ctx context.Context, addr string) error {
 
 	return nil
 }
+
+// ServePing exposes ping HTTP server to check the reachability between the
+// relay miner and its dependencies (Ex: relay server and their respective
+// backend URLs).
+func (rel *relayMiner) ServePing(ctx context.Context, network, addr string) error {
+	ln, err := net.Listen(network, addr)
+	if err != nil {
+		return err
+	}
+
+	// Start a long-lived goroutine that starts an HTTP server responding to
+	// ping requests. A single ping request on the relay server broadcasts a
+	// ping to all backing services/data nodes.
+	go func() {
+		if err := http.Serve(ln, rel.newPinghandlerFn(ctx, ln)); err != nil && !errors.Is(http.ErrServerClosed, err) {
+			rel.logger.Error().Err(err).Msg("ping server unexpectedly closed")
+		}
+	}()
+
+	go func() {
+		<-ctx.Done()
+		rel.logger.Info().Str("endpoint", addr).Msg("stopping ping server")
+		_ = ln.Close()
+	}()
+
+	return nil
+}
+
+func (rel *relayMiner) newPinghandlerFn(ctx context.Context, ln net.Listener) http.HandlerFunc {
```

Comment:
It seems like the `ln` variable is unused.
```suggestion
func (rel *relayMiner) newPinghandlerFn(ctx context.Context) http.HandlerFunc {
```